packages/react-instantsearch-core/src/core/__tests__/createInstantSearchManager.js

```diff
@@ -231,58 +250,55 @@ describe('createInstantSearchManager', () => {
     it('should be called when there is a new widget', () => {
       expect.assertions(2);
 
-      const client0 = makeClient(defaultResponse());
-      expect(client0.search).toHaveBeenCalledTimes(0);
+      const searchClient = createSearchClient();
+
       const ism = createInstantSearchManager({
         indexName: 'index',
-        initialState: {},
-        searchParameters: {},
-        searchClient: client0,
+        searchClient,
       });
 
       ism.widgetsManager.registerWidget({
         getMetadata: () => {},
         transitionState: () => {},
       });
 
+      expect(searchClient.search).toHaveBeenCalledTimes(0);
+
       return Promise.resolve().then(() => {
-        expect(client0.search).toHaveBeenCalledTimes(1);
+        expect(searchClient.search).toHaveBeenCalledTimes(1);
       });
     });
 
     it('should be called when there is a new client', () => {
       expect.assertions(4);
 
-      const client0 = makeClient(defaultResponse());
-      expect(client0.search).toHaveBeenCalledTimes(0);
+      const searchClient = createSearchClient();
+      const nextSearchClient = createSearchClient();
 
       const ism = createInstantSearchManager({
         indexName: 'index',
-        initialState: {},
-        searchParameters: {},
-        searchClient: client0,
+        searchClient,
       });
 
-      const client1 = makeClient(defaultResponse());
-      expect(client1.search).toHaveBeenCalledTimes(0);
+      expect(searchClient.search).toHaveBeenCalledTimes(0);
+      expect(nextSearchClient.search).toHaveBeenCalledTimes(0);
 
-      ism.updateClient(client1);
+      ism.updateClient(nextSearchClient);
 
       return Promise.resolve().then(() => {
-        expect(client0.search).toHaveBeenCalledTimes(0);
-        expect(client1.search).toHaveBeenCalledTimes(1);
+        expect(searchClient.search).toHaveBeenCalledTimes(0);
```

Comment:
Why is this promise.resolve needed ?
If I look at updateClient implem, it seems search method is called synchronously.
Is the .search method of the helper wrapper the client's search method?